src/display/annotation_layer.js

```diff
@@ -1832,157 +1844,216 @@ class ChoiceWidgetAnnotationElement extends WidgetAnnotationElement {
 
-  render() {
-    const BACKGROUND_ENLIGHT = 0.7;
+  #boundShow = this.#show.bind(this);
+
+  #boundToggle = this.#toggle.bind(this);
+
+  #color = null;
+
+  #container = null;
+
+  #contentsObj = null;
+
+  #elements = null;
+
+  #parent = null;
 
-    const wrapper = document.createElement("div");
-    wrapper.classList.add("popupWrapper");
+  #parentRect = null;
 
-    // For Popup annotations we hide the entire section because it contains
-    // only the popup. However, for Text annotations without a separate Popup
-    // annotation, we cannot hide the entire container as the image would
-    // disappear too. In that special case, hiding the wrapper suffices.
-    this.hideElement = this.hideWrapper ? wrapper : this.container;
-    this.hideElement.hidden = true;
+  #pinned = false;
 
-    const popup = document.createElement("div");
-    popup.classList.add("popup");
+  #popup = null;
 
-    const color = this.color;
-    if (color) {
-      // Enlighten the color.
-      const r = BACKGROUND_ENLIGHT * (255 - color[0]) + color[0];
-      const g = BACKGROUND_ENLIGHT * (255 - color[1]) + color[1];
-      const b = BACKGROUND_ENLIGHT * (255 - color[2]) + color[2];
-      popup.style.backgroundColor = Util.makeHexColor(r | 0, g | 0, b | 0);
+  #rect = null;
+
+  #richText = null;
+
+  #titleObj = null;
+
+  constructor({
+    container,
+    color,
+    elements,
+    titleObj,
+    modificationDate,
+    contentsObj,
+    richText,
+    parent,
+    rect,
+    parentRect,
+    open,
+  }) {
+    this.#container = container;
+    this.#titleObj = titleObj;
+    this.#contentsObj = contentsObj;
+    this.#richText = richText;
+    this.#parent = parent;
+    this.#color = color;
+    this.#rect = rect;
+    this.#parentRect = parentRect;
+    this.#elements = elements;
+
+    const dateObject = PDFDateString.toDateObject(modificationDate);
+    if (dateObject) {
+      // The modification date is shown in the popup instead of the creation
+      // date if it is available and can be parsed correctly, which is
+      // consistent with other viewers such as Adobe Acrobat.
+      this.#dateTimePromise = parent.l10n.get("annotation_date_string", {
+        date: dateObject.toLocaleDateString(),
+        time: dateObject.toLocaleTimeString(),
+      });
+    }
+
+    this.trigger = elements.flatMap(e => e.getElementsToTriggerPopup());
+    // Attach the event listeners to the trigger element.
+    for (const element of this.trigger) {
+      element.addEventListener("click", this.#boundToggle);
+      element.addEventListener("mouseenter", this.#boundShow);
+      element.addEventListener("mouseleave", this.#boundHide);
+      if (typeof PDFJSDev !== "undefined" && PDFJSDev.test("TESTING")) {
+        element.classList.add("popupTriggerArea");
+      }
+    }
+
+    this.#container.hidden = true;
+    if (open) {
+      this.#toggle();
+    }
+  }
+
+  static create(parameters) {
+    return new PopupElement(parameters);
+  }
```

Comment:
Sorry, but I don't quite understand why we need this static method and why can't we just do `new PopupElement(...)` inline at the only call-site?

Reply:
Because the linter is complaining:
```
 1866:5  error  Do not use 'new' for side effects  no-new
```

Reply:
Generally the ESLint rules are intended to catch things that could possibly lead to bugs, however in cases such as this it's fine to selectively disable individual rules when doing so makes sense.

So, please just remove this method and change the call-site into:
```javascript
// eslint-disable-next-line no-new
new PopupElement({
  ...
```